spec/controllers/v2/concerns/collection_spec.rb

```diff
@@ -0,0 +1,43 @@
+# frozen_string_literal: true
+
+# The `parents` parameter is required when testing nested controllers, and it should contain
+# an ordered list of reflection symbols similarly to how they are defined in the routes. It is
+# also required to set the `extra_params` variable in a let block and pass all the parent IDs
+# there as a hash. For example:
+# ```
+# it_behaves_like 'collection', :security_guide, :profile do
+#   let(:extra_params) { { security_guide_id: 123, profile_id: 456 } }
+# end
+# ```
+#
+# In a non-nested case this example can be simply included as:
+# ```
+# it_behaves_like 'collection'
+# ```
+#
+RSpec.shared_examples 'collection' do |*parents|
+  it 'returns base fields for each result' do
+    collection = items.map do |item|
+      hash_including(
+        'id' => item.id,
+        'type' => described_class.controller_name.chop,
+        'attributes' => attributes.each_with_object({}) do |(key, value), obj|
+          obj[key.to_s] = item.send(value)
+        end
+      )
+    end
+
+    if parents
+      get :index, params: { parents: parents }.merge(extra_params)
+    else
+      get :index
+    end
```

Comment:
I think this logic can be omitted if you look how existing examples deal with default values.